exchanges/binance/binance_wrapper.go

```diff
@@ -744,6 +744,9 @@ func (b *Binance) UpdateAccountInfo(ctx context.Context, assetType asset.Item) (
 		}
 		var currencyDetails []account.Balance
 		for i := range accData {
+			if acc.ID == "" {
+				acc.ID = accData[i].AccountAlias
```

Comment:
```json
Binance raw response: [{"accountAlias":"SgfWmYTiAuoC","asset":"DOT","balance":"0.00000000","crossWalletBalance":"0.00000000","crossUnPnl":"0.00000000","availableBalance":"0.00000000","maxWithdrawAmount":"0.00000000","marginAvailable":true,"updateTime":0},{"accountAlias":"SgfWmYTiAuoC","asset":"BTC","balance":"0.00000000","crossWalletBalance":"0.00000000","crossUnPnl":"0.00000000","availableBalance":"0.00000000","maxWithdrawAmount":"0.00000000","marginAvailable":true,"updateTime":0},{"accountAlias":"SgfWmYTiAuoC","asset":"SOL","balance":"0.00000000","crossWalletBalance":"0.00000000","crossUnPnl":"0.00000000","availableBalance":"0.00000000","maxWithdrawAmount":"0.00000000","marginAvailable":true,"updateTime":0},{"accountAlias":"SgfWmYTiAuoC","asset":"BNB","balance":"0.00000000","crossWalletBalance":"0.00000000","crossUnPnl":"0.00000000","availableBalance":"0.00000000","maxWithdrawAmount":"0.00000000","marginAvailable":true,"updateTime":0},{"accountAlias":"SgfWmYTiAuoC","asset":"ETH","balance":"0.00000000","crossWalletBalance":"0.00000000","crossUnPnl":"0.00000000","availableBalance":"0.00000000","maxWithdrawAmount":"0.00000000","marginAvailable":true,"updateTime":0},{"accountAlias":"SgfWmYTiAuoC","asset":"ADA","balance":"0.00000000","crossWalletBalance":"0.00000000","crossUnPnl":"0.00000000","availableBalance":"0.00000000","maxWithdrawAmount":"0.00000000","marginAvailable":true,"updateTime":0},{"accountAlias":"SgfWmYTiAuoC","asset":"USDT","balance":"0.00000000","crossWalletBalance":"0.00000000","crossUnPnl":"0.00000000","availableBalance":"0.00000000","maxWithdrawAmount":"0.00000000","marginAvailable":true,"updateTime":0},{"accountAlias":"SgfWmYTiAuoC","asset":"XRP","balance":"0.00000000","crossWalletBalance":"0.00000000","crossUnPnl":"0.00000000","availableBalance":"0.00000000","maxWithdrawAmount":"0.00000000","marginAvailable":true,"updateTime":0},{"accountAlias":"SgfWmYTiAuoC","asset":"BUSD","balance":"0.00000000","crossWalletBalance":"0.00000000","crossUnPnl":"0.00000000","availableBalance":"0.00000000","maxWithdrawAmount":"0.00000000","marginAvailable":true,"updateTime":0}]
```
This works if you only have one sub-account associated with those keys like the response above but will amalgamate multiple sub-accounts into one. Please change `var acc account.SubAccount` above to `var acc []account.SubAccount` then locally you can make a map[string aka AccountAlias][]account.Balance for example and then you can handle the entirety. 👨‍🎤 


Also out of scope; if you can just check the other asset types while you are there and update appropriately that would be awesome, if not I can check later.

Reply:
Yes make sense, added changes to separate balances into different buckets according to account id and a helper function to collect them back into a slice of accounts.